api/graphql/tests/test_reservation_units/test_reservation_unit_hauki_url.py

```diff
@@ -0,0 +1,162 @@
+import json
+
+import snapshottest
+from assertpy import assert_that
+from django.test import override_settings
+from freezegun import freeze_time
+
+from api.graphql.tests.base import GrapheneTestCaseBase
+from permissions.models import (
+    GeneralRoleChoice,
+    GeneralRolePermission,
+    ServiceSectorRoleChoice,
+    ServiceSectorRolePermission,
+    UnitRoleChoice,
+    UnitRolePermission,
+)
+from reservation_units.tests.factories import ReservationUnitFactory
+from spaces.tests.factories import ServiceSectorFactory, UnitFactory
+
+
+@override_settings(
+    HAUKI_SECRET="HAUKISECRET",
+    HAUKI_ADMIN_UI_URL="https://test.com",
+    HAUKI_ORIGIN_ID="origin",
+)
+@freeze_time("2021-05-03")
+class ReservationUnitHaukiUrlTestCase(GrapheneTestCaseBase, snapshottest.TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+
+        GeneralRolePermission.objects.create(
+            role=GeneralRoleChoice.objects.get(code="admin"),
+            permission="can_manage_units",
+        )
+
+        cls.unit = UnitFactory(tprek_department_id="depid")
+        cls.reservation_unit = ReservationUnitFactory(
+            unit=cls.unit, uuid="3774af34-9916-40f2-acc7-68db5a627710"
+        )
+        cls.target_runit = ReservationUnitFactory(
+            unit=cls.unit, uuid="3774af34-9916-40f2-acc7-68db5a627711"
+        )
+        cls.target_reservation_unit_ids = [str(cls.target_runit.id)]
+        cls.service_sector = ServiceSectorFactory(units=[cls.unit])
+
+    def get_query(self, res_unit_id=None, target_reservation_unit_ids=None):
+        target_reservation_unit_ids = (
+            target_reservation_unit_ids or self.target_reservation_unit_ids
+        )
+
+        target_res_units = ",".join([r for r in target_reservation_unit_ids])
+
+        res_unit_id = res_unit_id or self.reservation_unit.id
+
+        return """
+            query {
+                reservationUnitHaukiUrl(pk: %s reservationUnits: [%s]) {
+                    url
+                }
+            }
+        """ % (
+            res_unit_id,
+            target_res_units,
+        )
```

Comment:
Why not use f-strings?

Reply:
because of double {{ and }} does not go well with my taste 😂

Reply:
Sure, this is fine!